src/main/scala/esmeta/ir/util/Stringifier.scala

```diff
@@ -132,7 +132,12 @@ class Stringifier(detail: Boolean, location: Boolean) {
         field.map { case (t, f) => app >> ": " >> t >> " " >> f }
         app >> ")"
       case ESubstring(expr, from, to) =>
-        app >> "(substring " >> expr >> " " >> from >> " " >> to >> ")"
+        app >> "(substring " >> expr >> " " >> from
+        to match {
```

Comment:
Try to use the `map` function of the `Option[_]` type, or [For Comprehension](https://docs.scala-lang.org/tour/for-comprehensions.html). Do not use curly braces if possible.